libs/ledger/src/chaincode/smart_contract_manager.cpp

```diff
@@ -118,13 +119,14 @@ Contract::Result SmartContractManager::OnCreate(chain::Transaction const &tx)
   }
 
   nonce = FromBase64(nonce);
-  chain::Address const payable_address{crypto::Hash<crypto::SHA256>(tx.from().address() + nonce)};
+  chain::Address const contract_address{crypto::Hash<crypto::SHA256>(tx.from().address() + nonce)};
 
-  Identifier scope;
-  if (!scope.Parse(calculated_hash + "." + payable_address.display()))
+  SmartContractWrapper contract;
+  if (GetStateRecord(contract, contract_address.display()))
```

Comment:
leave address unencoded here

Reply:
As discussed, change Get- and SetStateRecord to accept Address argument, call display() inside